src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -371,6 +388,12 @@ struct TreeTraversal<BVH, Predicates, Callback, NearestPredicateTag>
         {
           if (HappyTreeFriends::isLeaf(_bvh, left_child))
           {
+            // Update distance with the distance to the actual primitive
+            auto primitive_index =
+                HappyTreeFriends::getLeafPermutationIndex(_bvh, left_child);
+            distance_left =
+                exactDistance(predicate, primitive_index, distance_left);
+
```

Comment:
This should probably be reworked. If a callback does not provide the distance, we should not compute the `primitive_index` as to not introduce a performance penalty.